### ISSUE TITLE
OperationalError: 14 UNAVAILABLE: DNS resolution failed in latest docker-compose with M1 Mac and Temporal v1.13

### ISSUE THREAD

**Issue by sw-yx** [temporalio/temporal#2117]
For those running the latest docker-compose, i am leaving a search trail for you to find this error, because this error is a very opaque one and what really happened is a segfault during bootstrap of docker compose.

## Expected Behavior

docker compose should work as expected

## Actual Behavior

You see this when you try to access Temporal Web

```
temporal-web            | [2021-11-01T04:45:09.772Z] OperationalError: 14 UNAVAILABLE: DNS resolution failed
temporal-web            |     at Object.exports.createStatusError (/usr/app/node_modules/grpc/src/common.js:91:15)
temporal-web            |     at Object.onReceiveStatus (/usr/app/node_modules/grpc/src/client_interceptors.js:1209:28)
temporal-web            |     at InterceptingListener._callNext (/usr/app/node_modules/grpc/src/client_interceptors.js:568:42)
temporal-web            |     at InterceptingListener.onReceiveStatus (/usr/app/node_modules/grpc/src/client_interceptors.js:618:8)
temporal-web            |     at callback (/usr/app/node_modules/grpc/src/client_interceptors.js:847:24) {
temporal-web            |   cause: Error: 14 UNAVAILABLE: DNS resolution failed
temporal-web            |       at Object.exports.createStatusError (/usr/app/node_modules/grpc/src/common.js:91:15)
temporal-web            |       at Object.onReceiveStatus (/usr/app/node_modules/grpc/src/client_interceptors.js:1209:28)
temporal-web            |       at InterceptingListener._callNext (/usr/app/node_modules/grpc/src/client_interceptors.js:568:42)
temporal-web            |       at InterceptingListener.onReceiveStatus (/usr/app/node_modules/grpc/src/client_interceptors.js:618:8)
temporal-web            |       at callback (/usr/app/node_modules/grpc/src/client_interceptors.js:847:24) {
temporal-web            |     code: 14,
temporal-web            |     metadata: Metadata { _internal_repr: {}, flags: 0 },
temporal-web            |     details: 'DNS resolution failed'
temporal-web            |   },
temporal-web            |   isOperational: true,
temporal-web            |   code: 14,
temporal-web            |   metadata: Metadata { _internal_repr: {}, flags: 0 },
temporal-web            |   details: 'DNS resolution failed'
temporal-web            | }
```

The real cause is higher up in the logs:

```
emporal                | PostgreSQL started.
temporal                | Setup PostgreSQL schema.
temporal                | + SCHEMA_DIR=/etc/temporal/schema/postgresql/v96/temporal/versioned
temporal                | + '[' temporal '!=' temporal ']'
temporal                | + temporal-sql-tool --plugin postgres --ep postgresql -u temporal -p 5432 --db temporal setup-schema -v 0.0
temporal                | qemu: uncaught target signal 11 (Segmentation fault) - core dumped
temporal                | ./auto-setup.sh: line 173:    27 Segmentation fault      temporal-sql-tool --plugin postgres --ep "${POSTGRES_SEEDS}" -u "${POSTGRES_USER}" -p "${DB_PORT}" --db "${DBNAME}" setup-schema -v 0.0
``` 

i sometimes also see this `relation "schema_version" already exists` error on subsequent runs:

```
temporal-postgresql     | ERROR:  relation "schema_version" already exists
temporal-postgresql     | STATEMENT:  CREATE TABLE schema_version(version_partition INT not null, db_name VARCHAR(255) not null, creation_time TIMESTAMP, curr_version VARCHAR(64), min_compatible_version VARCHAR(64), PRIMARY KEY (version_partition, db_name));
temporal                | 2021-11-01T04:50:44.866Z      ERROR   Unable to setup SQL schema.     {"error": "pq: relation \"schema_version\" already exists", "logging-call-at": "handler.go:57"}
temporal                | 2021/11/01 04:50:45 Loading config; env=docker,zone=,configDir=config
temporal                | 2021/11/01 04:50:45 Loading config files=[config/docker.yaml]
temporal                | {"level":"info","ts":"2021-11-01T04:50:45.279Z","msg":"Updated dynamic config","logging-call-at":"file_based_client.go:143"}
temporal                | {"level":"info","ts":"2021-11-01T04:50:45.282Z","msg":"Starting server for services","value":["history","matching","frontend","worker"],"logging-call-at":"server.go:123"}
temporal                | {"level":"info","ts":"2021-11-01T04:50:45.372Z","msg":"PProf not started due to port not set","logging-call-at":"pprof.go:67"}
temporal-postgresql     | ERROR:  duplicate key value violates unique constraint "namespaces_pkey"
temporal-postgresql     | DETAIL:  Key (partition_id, id)=(54321, \x32049b68787240948e63d0dd59896a83) already exists.
temporal-postgresql     | STATEMENT:  INSERT INTO 
temporal-postgresql     |        namespaces (partition_id, id, name, is_global, data, data_encoding, notification_version)
temporal-postgresql     |        VALUES($1, $2, $3, $4, $5, $6, $7)
```

## Steps to Reproduce the Problem

clone https://github.com/temporalio/docker-compose/commit/2f84de06e30be8dfeddc6f210ff2243ce9ddfe4d

and `docker-compose up` - should repro the error.

## Temporary resolution for those who run into this

`docker system prune -a && git reset 0e68621 --hard && docker-compose up` and the error is gone. 

## Specifications

  - Version: temporal 1.13
  - Platform: m1 mac mini

**Comment by sw-yx** [comment]
to maintainers - i dont know if this is the right place for this issue, but i couldnt create an issue in the docker compose repo. also happy to move it to the temporalio/web repo if it is related, but i suspect it is not, and temporal web is merely the symptom, not the cause.

**Comment by sw-yx** [comment]
note from Alex: "DNS resolution points to cgo. but I am not sure. We just need ARM images and it will mitigate all these problems."

**Comment by maxramqvist** [comment]
We see the same thing. 

Your official docker containers:
- temporal-server 1.14.2
- temporal-web 1.13

Linux x86 is the host. DNS lookup works fine inside the container, looking up the _TEMPORAL_GRPC_ENDPOINT_.

**Comment by maxramqvist** [comment]
@sw-yx So it seems the ARM images might not be the thing. Should I post a separate issue as this is not related to Macs or ARM?

**Comment by sw-yx** [comment]
this was resolved in 1.14.2
